packages/frontend/components/Plan/DeleteYearModal.tsx

```diff
@@ -0,0 +1,104 @@
+import { DeleteIcon } from "@chakra-ui/icons";
+import {
+  Button,
+  Flex,
+  IconButton,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { useRouter } from "next/router";
+import { Dispatch, SetStateAction } from "react";
+import { mutate } from "swr";
+import {
+  USE_STUDENT_WITH_PLANS_SWR_KEY,
+  useStudentWithPlans,
+} from "../../hooks";
+import { handleApiClientError, toast } from "../../utils";
+import { GraduateToolTip } from "../GraduateTooltip";
+
+interface DeleteYearModalProps {
+  yearNum: number;
+  removeYear: Dispatch<SetStateAction<number | undefined | null>>;
+}
+export const DeleteYearModal: React.FC<DeleteYearModalProps> = ({
+  yearNum,
+  removeYear,
+}) => {
+  const router = useRouter();
+  const { onOpen, onClose, isOpen } = useDisclosure();
+  const { student } = useStudentWithPlans();
+
+  if (!student) {
+    return <></>;
+  }
+
+  const deleteYear = async () => {
+    try {
+      // refresh the cache, show success message, and close the modal
+      mutate(USE_STUDENT_WITH_PLANS_SWR_KEY);
+      //   setSelectedPlanId(null);
```

Comment:
Is there a reason why this is commented out?

Reply:
Adressed in [7691ee0](https://github.com/sandboxnu/graduatenu/pull/689/commits/7691ee07923db58f72cb59a4d729f8c1cf2f4468).